### ISSUE TITLE
OpenGL recording of Carmageddon II running on wine does not work on Ubuntu Linux

### ISSUE THREAD

**Issue by juozaspo** [MaartenBaert/ssr#592]
## Description

Trying to record OpenGL of an old game running on wine results with error: `Could not get the size of the OpenGL application.`, in recent versions of wine, game also crash with an error in the NVidia driver file.

**OS:** Ubuntu versions 17.10,18.04
**OS Architecture:** 64bit
**SSR:** built from latest git source code
**Wine:** versions 2.22, 4.21; from winehq-devel package on winehq repository
**nGlide:** versions 1.05, 2.10; installed inside wine prefix, required for the game to run properly from the command line above.
**Video card:** GeForce 210 (with driver version 340.107)
**Desktop environment:** Xfce

Also after returning to desktop on game crash, the screen gamma is higher then it should be (caused by the game or the wine itself), resets upon upening nvidia settings progam.

**Update:** As of recent wine and ssr versions, recording works, but is buggy, as shown below. As it's not errorring like before any more.
**Update2:** The current wine-devel version (3.4) records ok, but old **outdated** staging version (2.21) still failing. Not sure with version fixes that, still needs testing.
**Update3:** Wine-staging branch is **discontinued**, as written in winehq.org, so closing this again.
**Update4:** Game running under recent winehq-devel versions from wine repository crashing when hitting report, with a log very similar to previous one, so I'm reopening this issue by adding more relevant information from #741.

<details><summary>Old SSR Log & configuration, Wine 2.22, nGlide 1.05</summary>

#### Configuration
Command line: `env WINEPREFIX=/home/juozas/.wine32 WINEARCH=win32 wine CARMA2_HW.exe`
Path: `/home/juozas/Wine/Carma2Mod`
#### Log
```
==================== SSR started ====================
SimpleScreenRecorder 0.3.8
Compiled with GCC 7.2.0
Qt: header 4.8.7, lib 4.8.7
libavformat: header 57.71.100, lib 57.71.100
libavcodec: header 57.89.100, lib 57.89.100
libavutil: header 55.58.100, lib 55.58.100
libswscale: header 4.6.100, lib 4.6.100
[CPUFeatures::Detect] CPU features: mmx sse sse2 sse3 ssse3
[ALSAInput::GetSourceList] Generating source list ...
[ALSAInput::GetSourceList] Found card: [hw:0] HDA Intel
[ALSAInput::GetSourceList] Found device: [hw:0,0] HDA Intel: ALC887 Analog
[ALSAInput::GetSourceList] Found device: [hw:0,2] HDA Intel: ALC887 Alt Analog
[ALSAInput::GetSourceList] Found card: [hw:1] HDA NVidia
[PulseAudioInput::GetSourceList] Generating source list ...
[SourceNamesCallback] Found source: [alsa_output.pci-0000_01_00.1.hdmi-stereo-extra1.monitor] Monitor of High Definition Audio Controller Digital Stereo (HDMI 2)
[SourceNamesCallback] Found source: [alsa_output.pci-0000_00_1b.0.analog-stereo.monitor] Monitor of Įtaisytas garsas Analoginė stereo
[SourceNamesCallback] Found source: [alsa_input.pci-0000_00_1b.0.analog-stereo] Įtaisytas garsas Analoginė stereo
[PageRecord::StartPage] Starting page ...
[PageRecord::StartPage] Started page.
[GLInjectInput::InputThread] Input thread started.
[SimpleSynth::Init] Warning: Buffer size 2048 is not supported, using 3072 instead. This is not a problem.
[SimpleSynth::SynthThread] Synth thread started.
[MakeThreadHighPriority] Warning: Can't increase the thread priority.
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Hooked: dlsym(glXGetProcAddressARB).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXSwapBuffers).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXCreateWindow).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXDestroyWindow).
[SSR-GLInject] [GLXFrameGrabber 1] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-33828058602-17553-glx0001-wine-preloader] Created video stream.
fixme:win:EnumDisplayDevicesW ((null),0,0x3282d4,0x00000000), stub!
fixme:d3d:debug_d3dformat Unrecognized 0x36314644 (as fourcc: DF16) WINED3DFORMAT!
fixme:d3d:wined3d_get_format Can't find format unrecognized (0x36314644) in the format lookup table.
fixme:d3d:debug_d3dformat Unrecognized 0x34324644 (as fourcc: DF24) WINED3DFORMAT!
fixme:d3d:wined3d_get_format Can't find format unrecognized (0x34324644) in the format lookup table.
[SSRVideoStreamWatcher::GetChanges] Added stream 33828058602-17553-glx0001-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSR-GLInject] [GLXFrameGrabber 2] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-33828329036-17553-glx0002-wine-preloader] Created video stream.
[SSR-GLInject] [GLXFrameGrabber 3] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-33828378037-17553-glx0003-wine-preloader] Created video stream.
fixme:win:EnumDisplayDevicesW ((null),0,0x3280f4,0x00000000), stub!
[SSRVideoStreamWatcher::GetChanges] Added stream 33828329036-17553-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSRVideoStreamWatcher::GetChanges] Added stream 33828378037-17553-glx0003-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSR-GLInject] OpenGL version = 3.3 (3.3.0 NVIDIA 340.104).
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-33828329036-17553-glx0002-wine-preloader] frame size = 1024x768.
[PageRecord::StartOutput] Starting output ...
[PageRecord::StartOutput] Error: Could not get the size of the OpenGL application. Either the application wasn't started correctly, or the application hasn't created an OpenGL window yet. If you want to start recording before starting the application, you have to enable scaling and enter the video size manually.
[PageRecord::StartOutput] Error: Something went wrong during initialization.
[SSR-GLInject] [GLXFrameGrabber 4] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-33833403044-17553-glx0004-wine-preloader] Created video stream.
[SSRVideoStreamWatcher::GetChanges] Added stream 33833403044-17553-glx0004-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
err:d3d:wined3d_unregister_window Window 0x10054 is not registered with wined3d.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 33833403044-17553-glx0004-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 33828378037-17553-glx0003-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 33828329036-17553-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 33828058602-17553-glx0001-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[PageRecord::StopPage] Stopping page ...
[GLInjectInput::~GLInjectInput] Stopping input thread ...
[GLInjectInput::InputThread] Input thread stopped.
[PageRecord::StopPage] Stopped page.
[SimpleSynth::~SimpleSynth] Stopping synth thread ...
[SimpleSynth::SynthThread] Synth thread stopped.
[SSR-GLInject] Library unloaded.
==================== SSR stopped ====================
```

</details>

<details><summary>Updated SSR Log & configuration, Wine 4.21, nGlide 2.10, with crash log included</summary>

#### Configuration
Command line: `wine CARMA2_HW.exe`
Path: `/home/juozas/Wine/Carma2`
#### Log 
```
$ LANGUAGE=C simplescreenrecorder 
==================== SSR started ====================
SimpleScreenRecorder 0.3.11
Compiled with GCC 7.4.0
Qt: header 5.9.5, lib 5.9.5
libavformat: header 57.83.100, lib 57.83.100
libavcodec: header 57.107.100, lib 57.107.100
libavutil: header 55.78.100, lib 55.78.100
libswscale: header 4.8.100, lib 4.8.100
[CPUFeatures::Detect] CPU features: mmx sse sse2 sse3 ssse3
[ALSAInput::GetSourceList] Generating source list ...
[ALSAInput::GetSourceList] Found card: [hw:0] HDA Intel
[ALSAInput::GetSourceList] Found device: [hw:0,0] HDA Intel: ALC887 Analog
[ALSAInput::GetSourceList] Found device: [hw:0,2] HDA Intel: ALC887 Alt Analog
[ALSAInput::GetSourceList] Found card: [hw:1] HDA NVidia
[PulseAudioInput::GetSourceList] Generating source list ...
[SourceNamesCallback] Found source: [alsa_output.pci-0000_01_00.1.hdmi-stereo-extra1.monitor] Monitor of High Definition Audio Controller Digital Stereo (HDMI 2)
[SourceNamesCallback] Found source: [alsa_output.pci-0000_00_1b.0.analog-stereo.monitor] Monitor of Įtaisytas garsas Analoginė stereo
[SourceNamesCallback] Found source: [alsa_input.pci-0000_00_1b.0.analog-stereo] Įtaisytas garsas Analoginė stereo
[PageRecord::StartPage] Starting page ...
[PageRecord::StartPage] Started page.
[GLInjectInput::InputThread] Input thread started.
[SimpleSynth::Init] Warning: Buffer size 2048 is not supported, using 3072 instead. This is not a problem.
[SimpleSynth::SynthThread] Synth thread started.
[MakeThreadHighPriority] Warning: Can't increase the thread priority.
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Hooked: dlsym(glXGetProcAddressARB).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXSwapBuffers).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXCreateWindow).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXDestroyWindow).
[SSR-GLInject] [GLXFrameGrabber 1] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-11433011974-14200-glx0001-wine-preloader] Created video stream.
0009:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-11433011974-14200-glx0001-wine-preloader] Destroyed video stream.
[SSR-GLInject] [GLXFrameGrabber 1] Destroyed GLX frame grabber.
0009:fixme:d3d:debug_d3dformat Unrecognized 0x36314644 (as fourcc: DF16) WINED3DFORMAT!
0009:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x36314644) in the format lookup table.
0009:fixme:d3d:debug_d3dformat Unrecognized 0x34324644 (as fourcc: DF24) WINED3DFORMAT!
0009:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x34324644) in the format lookup table.
[SSR-GLInject] [GLXFrameGrabber 2] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-11433092796-14200-glx0002-wine-preloader] Created video stream.
[SSRVideoStreamWatcher::GetChanges] Added stream 11433011974-14200-glx0001-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Error: Can't open video stream file!
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[GLInjectInput::SwitchStream] Error: Could not read stream, this usually means that the stream was already gone.
[SSRVideoStreamWatcher::GetChanges] Removed stream 11433011974-14200-glx0001-wine-preloader.
[SSRVideoStreamWatcher::GetChanges] Added stream 11433092796-14200-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSR-GLInject] [GLXFrameGrabber 3] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-11433164045-14200-glx0003-wine-preloader] Created video stream.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-11433164045-14200-glx0003-wine-preloader] Destroyed video stream.
[SSR-GLInject] [GLXFrameGrabber 3] Destroyed GLX frame grabber.
[SSRVideoStreamWatcher::GetChanges] Added stream 11433164045-14200-glx0003-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Error: Can't open video stream file!
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[GLInjectInput::SwitchStream] Error: Could not read stream, this usually means that the stream was already gone.
[SSRVideoStreamWatcher::GetChanges] Removed stream 11433164045-14200-glx0003-wine-preloader.
[SSR-GLInject] OpenGL version = 3.2 (3.2.0 NVIDIA 340.107).
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-11433092796-14200-glx0002-wine-preloader] frame size = 1366x768.
002f:fixme:d3d:state_linepattern_w Setting line patterns is not supported in OpenGL core contexts.
[PageRecord::StartOutput] Starting output ...
[Muxer::Init] Using format avi (AVI (Audio Video Interleaved)).
[Muxer::AddStream] Using codec libxvid (libxvidcore MPEG-4 part 2).
[VideoEncoder::PrepareStream] Using pixel format yuv420.
[Muxer::AddStream] Using codec libmp3lame (libmp3lame MP3 (MPEG audio layer 3)).
[AudioEncoder::PrepareStream] Using sample format f32p.
[BaseEncoder::EncoderThread] Encoder thread started.
[BaseEncoder::EncoderThread] Encoder thread started.
[Muxer::MuxerThread] Muxer thread started.
[PageRecord::StartOutput] Started output.
[Synchronizer::SynchronizerThread] Synchronizer thread started.
[PageRecord::StartInput] Starting input ...
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
[PulseAudioInput::InputThread] Stream is a monitor.
[PageRecord::StartInput] Started input.
[PulseAudioInput::InputThread] Input thread started.
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
[FastResampler::Resample] Resample ratio is 1.0000 (was 0.0000).
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPushClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopClientAttrib is deprecated and not available in preview contexts.".
002f:err:d3d:wined3d_debug_callback 0x21baf8: "GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.".
wine: Unhandled page fault on write access to 0AF107F0 at address F6787D92 (thread 002f), starting debugger...
[SSR-GLInject] Library loaded (32-bit).
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
Unhandled exception: page fault on write access to 0x0af107f0 in 32-bit code (0xf6787d92).
0033:fixme:dbghelp:elf_search_auxv can't find symbol in module
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:f6787d92 ESP:0745f850 EBP:00000200 EFLAGS:00010206(  R- --  I   - -P- )
 EAX:00000008 EBX:70c754c0 ECX:70c75500 EDX:0af107f0
 ESI:70c754c0 EDI:0af107f0
Stack dump:
0x0745f850:  0745f860 00000000 f679a0c0 00000002
0x0745f860:  0af107f0 00000000 00040004 f6279705
0x0745f870:  00000000 00000000 00000000 0745f9c0
0x0745f880:  00000001 00000200 70c754c0 f6261fbc
0x0745f890:  0af107f0 70c754c0 00000200 00000002
0x0745f8a0:  f679a0c0 00000000 0745f8f8 f7a17a9f
Backtrace:
=>0 0xf6787d92 _nv011glcore+0x98dd2() in libnvidia-glcore.so.340.107 (0x00000200)
  1 0xf6261fbc _nv011glcore+0xffffffff() in libnvidia-glcore.so.340.107 (0x70c754c0)
  2 0xf6267c1f _nv011glcore+0xffffffff() in libnvidia-glcore.so.340.107 (0xf6261f00)
  3 0xf626f4fd _nv011glcore+0xffffffff() in libnvidia-glcore.so.340.107 (0x70c71d20)
  4 0xf63aca9e _nv011glcore+0xffffffff() in libnvidia-glcore.so.340.107 (0x00000000)
  5 0xf63ad1f9 _nv011glcore+0xffffffff() in libnvidia-glcore.so.340.107 (0x0745fd38)
  6 0x7d1684b5 wined3d_texture_get_pitch+0x624() in wined3d (0x0745fd38)
  7 0x7d16bb01 wined3d_texture_update_desc+0x1a70() in wined3d (0x0745fdf8)
  8 0x7d164878 wined3d_swapchain_state_create+0x1ee7() in wined3d (0x0745fe58)
  9 0x7d1696cb wined3d_texture_get_pitch+0x183a() in wined3d (0x0745feb8)
  10 0x7d0f2893 wined3d_buffer_create+0xf392() in wined3d (0x0745fee8)
  11 0x7d0f4a91 wined3d_buffer_create+0x11590() in wined3d (0x0745ff38)
  12 0x7bcb29fc call_thread_func_wrapper+0xb() in ntdll (0x0745ff4c)
  13 0x7bcb620e RtlCaptureStackBackTrace+0x21d() in ntdll (0x0745ffdc)
  14 0x7bcb29ee call_thread_exit_func+0x31() in ntdll (0x0745ffec)
0xf6787d92 _nv011glcore+0x98dd2 in libnvidia-glcore.so.340.107: 	
Modules:
Module	Address			Debug info	Name (105 modules)
PE	  330000-  347000	Deferred        iforce2
PE	  400000-  7a1000	Deferred        carma2_hw
PE	 1a50000- 1f40000	Deferred        glide2x
PE	10000000-1001b000	Deferred        smackw32
PE	62340000-623bd000	Deferred        uxtheme
PE	62800000-628bb000	Deferred        usp10
PE	639c0000-63b08000	Deferred        winmm
PE	65200000-65679000	Deferred        ole32
PE	65980000-6599a000	Deferred        version
PE	66640000-66696000	Deferred        msacm32
PE	66cc0000-66eb5000	Deferred        ddraw
PE	6a300000-6a392000	Deferred        oleaut32
PE	6a900000-6aa74000	Deferred        setupapi
PE	6bac0000-6bae6000	Deferred        mcicda
PE	6c0c0000-6c0fb000	Deferred        imm32
PE	6c2c0000-6c2e3000	Deferred        wsock32
PE	6d5c0000-6d733000	Deferred        dsound
PE	6e8c0000-6ec8d000	Deferred        comctl32
PE	6fb40000-6fc76000	Deferred        d3d9
PE	6fdc0000-6ff8e000	Deferred        rpcrt4
PE	70340000-703b3000	Deferred        mmdevapi
PE	71240000-71463000	Deferred        kernelbase
ELF	7b400000-7b805000	Deferred        kernel32<elf>
  \-PE	7b420000-7b805000	\               kernel32
ELF	7bc00000-7beb3000	Dwarf           ntdll<elf>
  \-PE	7bc30000-7beb3000	\               ntdll
ELF	7c000000-7c004000	Deferred        <wine-loader>
ELF	7cf1c000-7d065000	Deferred        opengl32<elf>
  \-PE	7cf50000-7d065000	\               opengl32
ELF	7d065000-7d1f8000	Dwarf           wined3d<elf>
  \-PE	7d0b0000-7d1f8000	\               wined3d
ELF	7d997000-7d9af000	Deferred        libgpg-error.so.0
ELF	7d9af000-7d9c7000	Deferred        libresolv.so.2
ELF	7d9c7000-7da53000	Deferred        libvorbisenc.so.2
ELF	7da53000-7da7e000	Deferred        libvorbis.so.0
ELF	7da7e000-7da87000	Deferred        libogg.so.0
ELF	7da87000-7dae6000	Deferred        libflac.so.8
ELF	7dae6000-7dbc7000	Deferred        libgcrypt.so.20
ELF	7dbc7000-7dbdd000	Deferred        liblz4.so.1
ELF	7dbdd000-7dc09000	Deferred        liblzma.so.5
ELF	7dc09000-7dc10000	Deferred        libasyncns.so.0
ELF	7dc10000-7dc9f000	Deferred        libsndfile.so.1
ELF	7dc9f000-7dcaa000	Deferred        libwrap.so.0
ELF	7dcaa000-7dd3a000	Deferred        libsystemd.so.0
ELF	7dd3a000-7dd95000	Deferred        libdbus-1.so.3
ELF	7dd95000-7de22000	Deferred        libpulsecommon-11.1.so
ELF	7de22000-7de7c000	Deferred        libpulse.so.0
ELF	7de7c000-7deaf000	Deferred        winepulse<elf>
  \-PE	7de90000-7deaf000	\               winepulse
ELF	7e12c000-7e138000	Deferred        libxcursor.so.1
ELF	7e138000-7e14b000	Deferred        libxi.so.6
ELF	7e14b000-7e158000	Deferred        libxrandr.so.2
ELF	7e186000-7e241000	Deferred        winex11<elf>
  \-PE	7e1b0000-7e241000	\               winex11
ELF	7e2d5000-7e2d9000	Deferred        libxcomposite.so.1
ELF	7e2d9000-7e2e5000	Deferred        libxrender.so.1
ELF	7e2e5000-7e2ec000	Deferred        libxxf86vm.so.1
ELF	7e2ec000-7e2f0000	Deferred        libxinerama.so.1
ELF	7e3bb000-7e3ed000	Deferred        libexpat.so.1
ELF	7e3ed000-7e438000	Deferred        libfontconfig.so.1
ELF	7e438000-7e457000	Deferred        libz.so.1
ELF	7e457000-7e491000	Deferred        libpng16.so.16
ELF	7e491000-7e54e000	Deferred        libfreetype.so.6
ELF	7e54e000-7e571000	Deferred        libtinfo.so.5
ELF	7e571000-7e597000	Deferred        libncurses.so.5
ELF	7e5c5000-7e629000	Deferred        dinput<elf>
  \-PE	7e5d0000-7e629000	\               dinput
ELF	7e629000-7e746000	Deferred        ucrtbase<elf>
  \-PE	7e660000-7e746000	\               ucrtbase
ELF	7e746000-7e782000	Deferred        ws2_32<elf>
  \-PE	7e750000-7e782000	\               ws2_32
ELF	7e782000-7e85a000	Deferred        msvcrt<elf>
  \-PE	7e7b0000-7e85a000	\               msvcrt
ELF	7e85a000-7e8de000	Deferred        advapi32<elf>
  \-PE	7e870000-7e8de000	\               advapi32
ELF	7e8de000-7ea39000	Deferred        gdi32<elf>
  \-PE	7e900000-7ea39000	\               gdi32
ELF	7ea39000-7ec71000	Deferred        user32<elf>
  \-PE	7ea60000-7ec71000	\               user32
ELF	7efa3000-7efb7000	Deferred        libnss_files.so.2
ELF	7efb7000-7efd2000	Deferred        libnsl.so.1
ELF	7efd3000-7f000000	Deferred        iphlpapi<elf>
  \-PE	7efe0000-7f000000	\               iphlpapi
ELF	f5047000-f5062000	Deferred        libbsd.so.0
ELF	f5062000-f5069000	Deferred        libxdmcp.so.6
ELF	f5069000-f506d000	Deferred        libxau.so.6
ELF	f506d000-f516f000	Deferred        libm.so.6
ELF	f516f000-f5184000	Deferred        libxext.so.6
ELF	f5184000-f7718000	Dwarf           libnvidia-glcore.so.340.107
ELF	f7718000-f771d000	Deferred        libnvidia-tls.so.340.107
ELF	f771d000-f7749000	Deferred        libxcb.so.1
ELF	f7749000-f774e000	Deferred        libdl.so.2
ELF	f774e000-f776c000	Deferred        libgcc_s.so.1
ELF	f78f4000-f78fe000	Deferred        librt.so.1
ELF	f78fe000-f7978000	Deferred        libglu.so.1
ELF	f7978000-f7a8f000	Deferred        libgl.so.1
ELF	f7a8f000-f7a96000	Deferred        libxfixes.so.3
ELF	f7a96000-f7be0000	Deferred        libx11.so.6
ELF	f7be0000-f7dbc000	Deferred        libc.so.6
ELF	f7dbc000-f7ddb000	Deferred        libpthread.so.0
ELF	f7ddb000-f7f8e000	Dwarf           libwine.so.1
ELF	f7f8e000-f7fa1000	Deferred        libssr-glinject.so
ELF	f7fa2000-f7fb0000	Deferred        libnss_nis.so.2
ELF	f7fc5000-f7fcf000	Deferred        libnss_compat.so.2
ELF	f7fd1000-f7ff9000	Deferred        ld-linux.so.2
Threads:
process  tid      prio (all id:s are in hex)
00000008 (D) H:\Darbastalis\Carma2\CARMA2_HW.exe
	00000031   15
	00000030    0
	0000002f    0 <==
	0000002e   15
	0000002d   15
	0000002c    0
	0000002b    0
	00000009    0
0000000e services.exe
	00000024    0
	0000001d    0
	0000001a    0
	00000013    0
	00000010    0
	0000000f    0
00000011 plugplay.exe
	00000017    0
	00000016    0
	00000012    0
00000018 winedevice.exe
	00000021    0
	0000001e    0
	0000001c    0
	0000001b    0
	00000019    0
0000001f explorer.exe
	0000002a    0
	00000029    0
	00000028    0
	00000020    0
00000022 winedevice.exe
	00000027    0
	00000026    0
	00000025    0
	00000023    0
System information:
    Wine build: wine-4.21
    Platform: i386 (WOW64)
    Version: Windows XP
    Host system: Linux
    Host version: 4.15.0-72-generic
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 11433092796-14200-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[PageRecord::StopOutput] Stopping output ...
[PageRecord::StopOutput] Stopped output.
[PageRecord::StopInput] Stopping input ...
[PulseAudioInput::~PulseAudioInput] Stopping input thread ...
[PulseAudioInput::InputThread] Input thread stopped.
[PageRecord::StopInput] Stopped input.
[PageRecord::StopPage] Stopping page ...
[Synchronizer::~Synchronizer] Stopping synchronizer thread ...
[Synchronizer::SynchronizerThread] Synchronizer thread stopped.
[Muxer::~Muxer] Stopping encoders ...
[Muxer::~Muxer] Waiting for muxer thread to stop ...
[BaseEncoder::EncoderThread] Encoder thread stopped.
[BaseEncoder::EncoderThread] Encoder thread stopped.
[Muxer::MuxerThread] Muxer thread stopped.
[BaseEncoder::~BaseEncoder] Stopping encoder thread ...
[BaseEncoder::~BaseEncoder] Stopping encoder thread ...
[libmp3lame @ 0x55b22102a520] 3 frames left in the queue on closing
[GLInjectInput::~GLInjectInput] Stopping input thread ...
[GLInjectInput::InputThread] Input thread stopped.
[PageRecord::StopPage] Stopped page.
[SimpleSynth::~SimpleSynth] Stopping synth thread ...
[SimpleSynth::SynthThread] Synth thread stopped.
==================== SSR stopped ====================
[SSR-GLInject] Library unloaded.
```

</details>

**Comment by juozaspo** [comment]
Latest ssr build, as of f4f15171b38a94cae7f7a286b2954749ed95bcfd, still not working, not much difference in log.

<details><summary>ssr log from console including wine messages</summary>

```
==================== SSR started ====================
SimpleScreenRecorder 0.3.9
Compiled with GCC 7.2.0
Qt: header 5.9.1, lib 5.9.1
libavformat: header 57.71.100, lib 57.71.100
libavcodec: header 57.89.100, lib 57.89.100
libavutil: header 55.58.100, lib 55.58.100
libswscale: header 4.6.100, lib 4.6.100
[CPUFeatures::Detect] CPU features: mmx sse sse2 sse3 ssse3
[ALSAInput::GetSourceList] Generating source list ...
[ALSAInput::GetSourceList] Found card: [hw:0] HDA Intel
[ALSAInput::GetSourceList] Found device: [hw:0,0] HDA Intel: ALC887 Analog
[ALSAInput::GetSourceList] Found device: [hw:0,2] HDA Intel: ALC887 Alt Analog
[ALSAInput::GetSourceList] Found card: [hw:1] HDA NVidia
[PulseAudioInput::GetSourceList] Generating source list ...
[SourceNamesCallback] Found source: [alsa_output.pci-0000_00_1b.0.analog-stereo.monitor] Monitor of Įtaisytas garsas Analoginė stereo
[SourceNamesCallback] Found source: [alsa_input.pci-0000_00_1b.0.analog-stereo] Įtaisytas garsas Analoginė stereo
[PageRecord::StartPage] Starting page ...
[PageRecord::StartPage] Started page.
[GLInjectInput::InputThread] Input thread started.
[SimpleSynth::Init] Warning: Buffer size 2048 is not supported, using 3072 instead. This is not a problem.
[SimpleSynth::SynthThread] Synth thread started.
[MakeThreadHighPriority] Warning: Can't increase the thread priority.
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Hooked: dlsym(glXGetProcAddressARB).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXSwapBuffers).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXCreateWindow).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXDestroyWindow).
[SSR-GLInject] [GLXFrameGrabber 1] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-9306975917-8507-glx0001-wine-preloader] Created video stream.
0009:fixme:win:EnumDisplayDevicesW ((null),0,0x328324,0x00000000), stub!
0009:fixme:d3d:debug_d3dformat Unrecognized 0x36314644 (as fourcc: DF16) WINED3DFORMAT!
0009:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x36314644) in the format lookup table.
0009:fixme:d3d:debug_d3dformat Unrecognized 0x34324644 (as fourcc: DF24) WINED3DFORMAT!
0009:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x34324644) in the format lookup table.
[SSRVideoStreamWatcher::GetChanges] Added stream 9306975917-8507-glx0001-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSR-GLInject] [GLXFrameGrabber 2] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-9307322522-8507-glx0002-wine-preloader] Created video stream.
[SSRVideoStreamWatcher::GetChanges] Added stream 9307322522-8507-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSR-GLInject] [GLXFrameGrabber 3] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-9307384237-8507-glx0003-wine-preloader] Created video stream.
0009:fixme:win:EnumDisplayDevicesW ((null),0,0x328144,0x00000000), stub!
[SSR-GLInject] OpenGL version = 3.3 (3.3.0 NVIDIA 340.104).
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-9307322522-8507-glx0002-wine-preloader] frame size = 1024x768.
[SSRVideoStreamWatcher::GetChanges] Added stream 9307384237-8507-glx0003-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[PageRecord::StartOutput] Starting output ...
[PageRecord::StartOutput] Error: Could not get the size of the OpenGL application. Either the application wasn't started correctly, or the application hasn't created an OpenGL window yet. If you want to start recording before starting the application, you have to enable scaling and enter the video size manually.
[PageRecord::StartOutput] Error: Something went wrong during initialization.
[SSR-GLInject] [GLXFrameGrabber 4] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-9320264721-8507-glx0004-wine-preloader] Created video stream.
[SSRVideoStreamWatcher::GetChanges] Added stream 9320264721-8507-glx0004-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
0009:err:d3d:wined3d_unregister_window Window 0x10054 is not registered with wined3d.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 9320264721-8507-glx0004-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 9307384237-8507-glx0003-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 9307322522-8507-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamWatcher::Init] Deleted abandoned stream 9306975917-8507-glx0001-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSR-GLInject] Library unloaded.
[PageRecord::StopPage] Stopping page ...
[GLInjectInput::~GLInjectInput] Stopping input thread ...
[GLInjectInput::InputThread] Input thread stopped.
[PageRecord::StopPage] Stopped page.
[SimpleSynth::~SimpleSynth] Stopping synth thread ...
[SimpleSynth::SynthThread] Synth thread stopped.
==================== SSR stopped ====================
```

</details>

**Comment by kn00tcn** [comment]
does it launch a second process or something? does any other wine-ogl app do this? did you enable the scale button & input your resolution manually?

**Comment by juozaspo** [comment]
The game screen launches directly, skipping game launcher, in full screen. The game needs **3dfx/glide** libraries installed for it to work in it's default mode, those dll's are provided by **nglide** software as posted above. It renders to **direct3d** where wine translates it to **opengl**. 
Didn't test other directx game, will try later.
No scale options enabled in ssr settings, no manual resolution selected.
Launch manually not practical, game runs in full screen, alt-tab'ing it glitches it's controls out in some wine configurations, can't get back to it later. So I always launch it automatically.
I'll update ssr to latest build and do some testing, will post results later after I'm done.

**Comment by juozaspo** [comment]
Update: recording partially works, errors out doing game launch, but recording saves ok, as of recent wine version (3.4). The screen size is 4x3, recorded from old crt monitor, the other monitor not working atm, will test when that monitor gets fixed. 

<details><summary>Recording from wine 3.4, partially working</summary>

```
juozas@xubuntu:~$ LANGUAGE=en simplescreenrecorder 
==================== SSR started ====================
SimpleScreenRecorder 0.3.10
Compiled with GCC 7.2.0
Qt: header 5.9.1, lib 5.9.1
libavformat: header 57.71.100, lib 57.71.100
libavcodec: header 57.89.100, lib 57.89.100
libavutil: header 55.58.100, lib 55.58.100
libswscale: header 4.6.100, lib 4.6.100
[CPUFeatures::Detect] CPU features: mmx sse sse2 sse3 ssse3
[ALSAInput::GetSourceList] Generating source list ...
[ALSAInput::GetSourceList] Found card: [hw:0] HDA Intel
[ALSAInput::GetSourceList] Found device: [hw:0,0] HDA Intel: ALC887 Analog
[ALSAInput::GetSourceList] Found device: [hw:0,2] HDA Intel: ALC887 Alt Analog
[ALSAInput::GetSourceList] Found card: [hw:1] HDA NVidia
[PulseAudioInput::GetSourceList] Generating source list ...
[SourceNamesCallback] Found source: [alsa_output.pci-0000_00_1b.0.analog-stereo.monitor] Monitor of Įtaisytas garsas Analoginė stereo
[SourceNamesCallback] Found source: [alsa_input.pci-0000_00_1b.0.analog-stereo] Įtaisytas garsas Analoginė stereo
[PageRecord::StartPage] Starting page ...
[PageRecord::StartPage] Started page.
[GLInjectInput::InputThread] Input thread started.
[SimpleSynth::Init] Warning: Buffer size 2048 is not supported, using 3072 instead. This is not a problem.
[SimpleSynth::SynthThread] Synth thread started.
[MakeThreadHighPriority] Warning: Can't increase the thread priority.
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (32-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Library loaded (64-bit).
[SSR-GLInject] Hooked: dlsym(glXGetProcAddressARB).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXSwapBuffers).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXCreateWindow).
[SSR-GLInject] Hooked: glXGetProcAddressARB(glXDestroyWindow).
[SSR-GLInject] [GLXFrameGrabber 1] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7071623697-5319-glx0001-wine-preloader] Created video stream.
0009:fixme:win:EnumDisplayDevicesW ((null),0,0x328324,0x00000000), stub!
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7071623697-5319-glx0001-wine-preloader] Destroyed video stream.
[SSR-GLInject] [GLXFrameGrabber 1] Destroyed GLX frame grabber.
0009:fixme:d3d:debug_d3dformat Unrecognized 0x36314644 (as fourcc: DF16) WINED3DFORMAT!
0009:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x36314644) in the format lookup table.
0009:fixme:d3d:debug_d3dformat Unrecognized 0x34324644 (as fourcc: DF24) WINED3DFORMAT!
0009:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x34324644) in the format lookup table.
[SSRVideoStreamWatcher::GetChanges] Added stream 7071623697-5319-glx0001-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Error: Can't open video stream file!
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[GLInjectInput::SwitchStream] Error: Could not read stream, this usually means that the stream was already gone.
[SSRVideoStreamWatcher::GetChanges] Removed stream 7071623697-5319-glx0001-wine-preloader.
[SSR-GLInject] [GLXFrameGrabber 2] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7071896206-5319-glx0002-wine-preloader] Created video stream.
[SSRVideoStreamWatcher::GetChanges] Added stream 7071896206-5319-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSR-GLInject] [GLXFrameGrabber 3] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7072025011-5319-glx0003-wine-preloader] Created video stream.
0009:fixme:win:EnumDisplayDevicesW ((null),0,0x328144,0x00000000), stub!
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7072025011-5319-glx0003-wine-preloader] Destroyed video stream.
[SSR-GLInject] [GLXFrameGrabber 3] Destroyed GLX frame grabber.
[SSRVideoStreamWatcher::GetChanges] Added stream 7072025011-5319-glx0003-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Error: Can't open video stream file!
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[GLInjectInput::SwitchStream] Error: Could not read stream, this usually means that the stream was already gone.
[SSRVideoStreamWatcher::GetChanges] Removed stream 7072025011-5319-glx0003-wine-preloader.
[SSR-GLInject] OpenGL version = 3.3 (3.3.0 NVIDIA 340.104).
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7071896206-5319-glx0002-wine-preloader] frame size = 1024x768.
[PageRecord::StartOutput] Starting output ...
[Muxer::Init] Using format avi (AVI (Audio Video Interleaved)).
[Muxer::AddStream] Using codec libxvid (libxvidcore MPEG-4 part 2).
[VideoEncoder::PrepareStream] Using pixel format yuv420.
[BaseEncoder::EncoderThread] Encoder thread started.
[Muxer::AddStream] Using codec libmp3lame (libmp3lame MP3 (MPEG audio layer 3)).
[AudioEncoder::PrepareStream] Using sample format f32p.
[libmp3lame @ 0x5633975a8c80] Channel layout not specified
[BaseEncoder::EncoderThread] Encoder thread started.
[PageRecord::StartOutput] Started output.
[PageRecord::StartInput] Starting input ...
[Synchronizer::SynchronizerThread] Synchronizer thread started.
[Muxer::MuxerThread] Muxer thread started.
[PulseAudioInput::InputThread] Stream is a monitor.
[PageRecord::StartInput] Started input.
[PulseAudioInput::InputThread] Input thread started.
[FastResampler::Resample] Resample ratio is 1.0000 (was 0.0000).
[PageRecord::StopOutput] Stopping output ...
[Synchronizer::~Synchronizer] Stopping synchronizer thread ...
[Synchronizer::SynchronizerThread] Synchronizer thread stopped.
[Muxer::Finish] Finishing encoders ...
[BaseEncoder::EncoderThread] Flushing encoder ...
[BaseEncoder::EncoderThread] Encoder thread stopped.
[BaseEncoder::EncoderThread] Encoder thread stopped.
[Muxer::MuxerThread] Muxer thread stopped.
[Muxer::~Muxer] Stopping encoders ...
[Muxer::~Muxer] Waiting for muxer thread to stop ...
[BaseEncoder::~BaseEncoder] Stopping encoder thread ...
[BaseEncoder::~BaseEncoder] Stopping encoder thread ...
[PageRecord::StopOutput] Stopped output.
[PageRecord::StopInput] Stopping input ...
[PulseAudioInput::~PulseAudioInput] Stopping input thread ...
[PulseAudioInput::InputThread] Input thread stopped.
[PageRecord::StopInput] Stopped input.
[SSR-GLInject] [GLXFrameGrabber 4] Created GLX frame grabber.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7163498501-5319-glx0004-wine-preloader] Created video stream.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7163498501-5319-glx0004-wine-preloader] Destroyed video stream.
[SSR-GLInject] [GLXFrameGrabber 4] Destroyed GLX frame grabber.
[SSRVideoStreamWatcher::GetChanges] Added stream 7163498501-5319-glx0004-wine-preloader.
[SSRVideoStreamReader::Init] Created video stream reader.
[SSRVideoStreamReader::Init] Error: Can't open video stream file!
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[GLInjectInput::SwitchStream] Error: Could not read stream, this usually means that the stream was already gone.
[SSRVideoStreamWatcher::GetChanges] Removed stream 7163498501-5319-glx0004-wine-preloader.
0009:err:d3d:wined3d_unregister_window Window 0x10054 is not registered with wined3d.
[SSR-GLInject] [/dev/shm/ssr-channel-juozas/video-7071896206-5319-glx0002-wine-preloader] Destroyed video stream.
[SSR-GLInject] [GLXFrameGrabber 2] Destroyed GLX frame grabber.
[SSRVideoStreamWatcher::GetChanges] Removed stream 7071896206-5319-glx0002-wine-preloader.
[SSRVideoStreamReader::Init] Destroyed video stream reader.
[SSR-GLInject] Library unloaded.
[PageRecord::StopPage] Stopping page ...
[GLInjectInput::~GLInjectInput] Stopping input thread ...
[GLInjectInput::InputThread] Input thread stopped.
[PageRecord::StopPage] Stopped page.
[SimpleSynth::~SimpleSynth] Stopping synth thread ...
[SimpleSynth::SynthThread] Synth thread stopped.
==================== SSR stopped ====================
juozas@xubuntu:~$ wine --version 
wine-3.4
```

</details>

**Comment by kn00tcn** [comment]
almost forgot, what about wine-staging? since regular wine prefers generic implementations while staging does more game specific or gfx driver quirk specific fixes

**Comment by juozaspo** [comment]
My version of wine is from wine-devel package. Might try staging, but I don't expect much different results.

**Comment by juozaspo** [comment]
Recording still fail with old wine versions such as wine-staging (2.21), while wine-devel (3.4) partially working.

**Comment by juozaspo** [comment]
It appears recording does not work in recent wine versions on Ubuntu 18.04 with legacy Nvidia driver, so I'm reopening this.

**Comment by juozaspo** [comment]
Game does not crash when running in Direct3D mode under wine, but then it is locked to 640x480 resolution (by game design) so only the default configuration using nGlide as Glide wrapper is affected. As the game launcher (`carma2.exe`) running on wine always crash right after showing it's dialog, the game must be started by calling it's executable directly whitch works if the glide wrapper is already is installed (the game executable does not use direct3d by default, and the laucher prefers Glide even if Direct3D is available).

**Comment by MaartenBaert** [comment]
This sounds like an internal incompatibility between the various emulation layers you are using to get graphics to work, and the injection mechanism SSR uses to capture OpenGL. I don't think there's much I can do about this since OpenGL recording is really just a hack, and I suspect it's just not compatible with what WINE is doing here.
